### PR TITLE
test: add e2e test for Thanos Querier's kube-rbac-proxy

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

**Comment by simonpasquier** [comment]
This adds an end-to-end to validate the kube-rbac-proxy -> prom-label-proxy -> Thanos querier path for queries. I'm doing this before adding the same for rules and alerts.

cc @openshift/openshift-team-monitoring

File: test/e2e/alertmanager_test.go [hunk 1]
```diff
@@ -270,27 +269,17 @@ func TestAlertmanagerKubeRbacProxy(t *testing.T) {
 		}
 
 		err = framework.Poll(5*time.Second, 5*time.Minute, func() error {
-			secrets, err := f.KubeClient.CoreV1().Secrets(testNs).List(metav1.ListOptions{})
+			token, err := f.GetServiceAccountToken(testNs, sa)
```
Comment (s-urbaniak) [inline test/e2e/alertmanager_test.go:272]: nice :+1:

File: test/e2e/framework/client.go [hunk 2]
```diff
@@ -20,69 +20,68 @@ import (
...
-
-	for _, secret := range secrets.Items {
-		_, dockerToken := secret.Annotations["openshift.io/create-dockercfg-secrets"]
-		e2eToken := strings.Contains(secret.Name, "cluster-monitoring-operator-e2e-token-")
-
-		// we have to skip the token secret that contains the openshift.io/create-dockercfg-secrets annotation
-		// as this is the token to talk to the internal registry.
-		if !dockerToken && e2eToken {
-			token = string(secret.Data["token"])
+func (c *PrometheusClient) injectQueryParameters(req *http.Request, kvs ...string) {
```
Comment (s-urbaniak) [inline test/e2e/framework/client.go:68]: nice API surface :+1:

**Pull request merged**